### PR TITLE
Provide RPM support

### PR DESCRIPTION
**Pull request by konstantina-gramatova**
[#85] Provide RPM support

Added RPM packaging and relevant resources - scripts, builds, packaging config.

Signed-off-by: Konstantina Gramatova <konstantina.gramatova@bosch.io>

### CODE REVIEW

**Comment by dimiter-todorov-georgiev** [comment]
The build with goreleaser passes. The RPM successfully deploys on Fedora-IOT x64 and the DEB file successfully deploys on Raspbian/arm64.

**Comment by konstantina-gramatova** [comment]
Thank you @dimiter-todorov!

File: build/prerm_rpm [hunk 1]
```diff
@@ -0,0 +1,18 @@
+if [ $1 -eq 0 ] && [ -x "/usr/lib/systemd/systemd-update-helper" ]; then
+    # Package removal, not upgrade
```
Comment (e-grigorov) [inline build/prerm_rpm:2]: On removal, the suite connector mosquitto configuration is removed, I guess that the mosquitto service has to be restarted.

File: build/postinst_rpm [hunk 2]
```diff
@@ -0,0 +1,24 @@
+if [ $1 -eq 1 ] && [ -x "/usr/lib/systemd/systemd-update-helper" ]; then
+    # Initial installation
+
+    # Configure Mosquitto to look for the suite-connector config file
+    echo "include_dir /etc/mosquitto/conf.d" >> /etc/mosquitto/mosquitto.conf
```
Comment (e-grigorov) [inline build/postinst_rpm:5]: If mosquitto service is already there, do we need explicit restart to apply the new configuration?

File: build/postinst_rpm [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    # as per Fedora's guidelines
+
+    # Use `:` instead of `true` as this is the usual expansion of rpm scriptlets
+
+    # containerd and mosquitto
+    /usr/lib/systemd/systemd-update-helper install-system-units containerd.service || :
+    /usr/lib/systemd/systemd-update-helper install-system-units mosquitto.service || :
+
+    # The Kanto services
+    /usr/lib/systemd/systemd-update-helper install-system-units suite-connector.service || :
```
Comment (e-grigorov) [inline build/postinst_rpm:18]: If enabled, can we auto-start them? What's the RPM expected behavior?

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: build/postinst_rpm (hunk 3 above)
Comment (dimiter-todorov-georgiev) [reply build/postinst_rpm:18]: It's expected not to start them.

File: build/postinst_rpm (hunk 2 above)
Comment (dimiter-todorov-georgiev) [reply build/postinst_rpm:5]: Yes, I think so.

File: build/prerm_rpm (hunk 1 above)
Comment (dimiter-todorov-georgiev) [reply build/prerm_rpm:2]: Maybe so. I think it should be able to keep running with the existing configuration, even though the files have been removed, until the next time that mosquitto is restarted. 

Maybe we should restart it here.

On the other hand, maybe we shouldn't touch a service that isn't getting uninstalled right now.

File: build/postinst_rpm (hunk 3 above)
Comment (e-grigorov) [reply build/postinst_rpm:18]: Basically, this means that they will not be started right after the installation but after the first system reboot. It sounds strange, is there a doc reference?

File: build/prerm_rpm (hunk 1 above)
Comment (e-grigorov) [reply build/prerm_rpm:2]: But the service is going to operate with obsolete configuration. In any case, it seems that Debian package does so: https://github.com/eclipse-kanto/kanto/blob/main/build/postrm

File: build/postinst_rpm (hunk 3 above)
Comment (dimiter-todorov-georgiev) [reply build/postinst_rpm:21]: I'm not sure if there's an actual doc, but they are not suggesting that .preset files are distributed in the same rpm as the services, which we're doing here.

File: build/prerm_rpm (hunk 1 above)
Comment (dimiter-todorov-georgiev) [reply build/prerm_rpm:2]: Done

File: build/postinst_rpm (hunk 2 above)
Comment (dimiter-todorov-georgiev) [reply build/postinst_rpm:5]: Done

File: build/prerm_rpm (hunk 1 above)
Comment (e-grigorov) [reply build/prerm_rpm:2]: > Done

What do you mean? Mosquitto is still not restarted in the post remove script.

File: build/postinst_rpm (hunk 3 above)
Comment (e-grigorov) [reply build/postinst_rpm:21]: I agree about the preset but I am asking about auto-start if enabled. Is it possible to do so with the current helpers? If marked for restart or?

File: build/postinst_rpm (hunk 2 above)
Comment (e-grigorov) [reply build/postinst_rpm:5]: > Done

Now, the suite-connector is restarted but my comment was about the mosquitto. Why is it?

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: build/postinst_rpm (hunk 2 above)
Comment (dimiter-todorov-georgiev) [reply build/postinst_rpm:5]: This was a mistake, sorry!

File: build/postinst_rpm (hunk 3 above)
Comment (dimiter-todorov-georgiev) [reply build/postinst_rpm:21]: We will check if the service is enabled using the systemctl command, and then start the service if it's enabled.

File: build/postinst_rpm (hunk 2 above)
Comment (dimiter-todorov-georgiev) [reply build/postinst_rpm:5]: This was an error. Will be fixed.

File: build/prerm_rpm (hunk 1 above)
Comment (dimiter-todorov-georgiev) [reply build/prerm_rpm:2]: This was an error. Will be fixed.

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**